### PR TITLE
feat: Search highlighting for Collectives

### PR DESCRIPTION
**Pull request by elzody**
### 📝 Summary

* Resolves: #648 

This pull request integrates search highlighting from the text app into collectives.

#### 🖼️ Video

https://github.com/user-attachments/assets/c8614202-f057-464f-94ed-de05b38249ef

### 🚧 TODO

- [ ] Tests, will be implemented later and tracked here #1381

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] Tests (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [ ] Documentation ([README](https://github.com/nextcloud/collectives/blob/main/README.md) or [documentation](https://github.com/nextcloud/collectives/blob/main/docs/)) has been updated or is not required

### CODE REVIEW

**Comment by nimishavijay** [comment]
Looks really nice overall! Some comments from my side: 

- [ ] "Find prev" --> "Find previous" to be clear to non-native English speakers as well as to avoid translation issues
- [ ] add a few pixels more margin to the left of "Found 34 matches" 
- [ ] if possible, we could also show the search string in the bottom bar by changing the wording like `Found 34 matches for "she"`
- [ ] Usually the icons for buttons are on the left, so we should follow that pattern here too in the "Find next" and "Find previous" buttons
- [ ] the "Find all" toggle appears as a button and that may be an accessibility issue, so we can change that to a checkbox element and change the wording to "Highlight all matches" (ref Firefox's design)
- [ ] it would also be nice to have a tertiary "Clear search" button on the left of "Find previous" (it could have a `X` icon) 


Other than that, it looks really good! :)

**Review by jancborchardt — DISMISSED** [review]
Nice! Only additional feedback in addition to @nimishavijay’s is:
- [ ] The elements in the left navigation directly below the search field jump around vertically a little bit. It would be nice if they stay in place, e.g. the main Collective page.

**Comment by juliushaertl** [comment]
> Nice! Only additional feedback in addition to @nimishavijay’s is:
> 
> - [ ] The elements in the left navigation directly below the search field jump around vertically a little bit. It would be nice if they stay in place, e.g. the main Collective page.

Let's file that into a separate issue. Unrelated to this part of the implementation

**Review by AndyScherzinger — APPROVED** [review]

File: src/components/SearchDialog.vue [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	},
+}
+</script>
+
+<style lang="scss" scoped>
+$button-gap: calc(var(--default-grid-baseline) * 3);
+
+.search-dialog__container {
+	width: 100%;
+	height: 50px;
```
Comment (mejo-) [inline src/components/SearchDialog.vue:133]: I would remove the `height` property altogether. With the `margin-bottom` and `margin-top` suggested fo rthe highlight button below, it's guaranteed that it's high enough. And on narrow screens where buttons become two lines high, `50px` are not enough.

**Review by mejo- — APPROVED** [review]

**Pull request merged**